cloud/ec2.go

```diff
@@ -297,71 +278,112 @@ func (m *ec2Manager) spawnOnDemandHost(ctx context.Context, h *host.Host, ec2Set
 		}
 	} else {
-		spotRequest.LaunchSpecification.SecurityGroups = ec2Settings.getSecurityGroups()
+		launchTemplate.SecurityGroups = ec2Settings.getSecurityGroups()
+	}
+
+	userData, err := bootstrapUserData(ctx, evergreen.GetEnvironment(), h, ec2Settings.UserData)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not add bootstrap script to user data")
 	}
+	ec2Settings.UserData = userData
 
 	if ec2Settings.UserData != "" {
-		expanded, err := m.expandUserData(ec2Settings.UserData)
+		var expanded string
+		expanded, err = m.expandUserData(ec2Settings.UserData)
 		if err != nil {
 			return nil, errors.Wrap(err, "problem expanding user data")
 		}
 		userData := base64.StdEncoding.EncodeToString([]byte(expanded))
-		spotRequest.LaunchSpecification.UserData = &userData
+		launchTemplate.UserData = &userData
 	}
 
-	userData, err := bootstrapUserData(ctx, evergreen.GetEnvironment(), h, ec2Settings.UserData)
+	createTemplateResponse, err := m.client.CreateLaunchTemplate(ctx, &ec2.CreateLaunchTemplateInput{
+		LaunchTemplateData: launchTemplate,
+		// mandatory field may only contain letters, numbers, and the following characters: - ( ) . / _
+		LaunchTemplateName: aws.String(fmt.Sprintf("%s", util.RandomString())),
+	})
 	if err != nil {
-		return nil, errors.Wrap(err, "could not add bootstrap script to user data")
+		return nil, errors.Wrap(err, "can't upload config template to AWS")
+	}
+	err = validateCreateTemplateResponse(createTemplateResponse)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid create template response")
+	}
+	templateID = createTemplateResponse.LaunchTemplate.LaunchTemplateId
+
+	// Create a fleet with a single spot instance from the launch template
+	createFleetInput := &ec2.CreateFleetInput{
+		LaunchTemplateConfigs: []*ec2.FleetLaunchTemplateConfigRequest{
+			{
+				LaunchTemplateSpecification: &ec2.FleetLaunchTemplateSpecificationRequest{
+					LaunchTemplateId: templateID,
+					Version:          aws.String(strconv.Itoa(int(*createTemplateResponse.LaunchTemplate.LatestVersionNumber))),
+				},
+			},
+		},
+		TargetCapacitySpecification: &ec2.TargetCapacitySpecificationRequest{
+			TotalTargetCapacity:       aws.Int64(1),
+			DefaultTargetCapacityType: aws.String("spot"),
+		},
+		Type: aws.String("instant"),
+	}
+	createFleetResponse, err := m.client.CreateFleet(ctx, createFleetInput)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating fleet")
+	}
+	err = validateCreateFleetResponse(createFleetResponse)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid create fleet response")
 	}
-	ec2Settings.UserData = userData
 
-	grip.Debug(message.Fields{
-		"message":       "starting spot instance",
-		"args":          spotRequest,
-		"host":          h.Id,
-		"host_provider": h.Distro.Provider,
-		"distro":        h.Distro.Id,
-	})
-	spotResp, err := m.client.RequestSpotInstances(ctx, spotRequest)
+	h.Id = *createFleetResponse.Instances[0].InstanceIds[0]
```

Comment:
Is it possible for the fleet response to _not_ have values at `Instances[0]` or `InstanceIds[0]`?

Reply:
`validateCreateFleetResponse` makes sure this is not the case.